contracts/router/modules/bridge/SynapseBridgeModule.sol

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {OnlyDelegateCall} from "../pool/OnlyDelegateCall.sol";
+import {IBridgeModule} from "../../interfaces/IBridgeModule.sol";
+import {ILocalBridgeConfig} from "../../interfaces/ILocalBridgeConfig.sol";
+import {ISynapseBridge} from "../../interfaces/ISynapseBridge.sol";
+import {Action, BridgeToken, DefaultParams, SwapQuery} from "../../libs/Structs.sol";
+import {UniversalTokenLib} from "../../libs/UniversalToken.sol";
+
+contract SynapseBridgeModule is OnlyDelegateCall, IBridgeModule {
+    using UniversalTokenLib for address;
+
+    error SynapseBridgeModule__EqualSwapIndexes(uint8 index);
+    error SynapseBridgeModule__UnsupportedDepositAction(Action action);
+    error SynapseBridgeModule__UnsupportedRedeemAction(Action action);
+    error SynapseBridgeModule__UnsupportedToken(address token);
+
+    /// These need to be immutable in order to be accessed via delegatecall
+    ILocalBridgeConfig public immutable localBridgeConfig;
+    ISynapseBridge public immutable synapseBridge;
+
+    constructor(address localBridgeConfig_, address synapseBridge_) {
+        localBridgeConfig = ILocalBridgeConfig(localBridgeConfig_);
+        synapseBridge = ISynapseBridge(synapseBridge_);
+    }
+
+    /// @inheritdoc IBridgeModule
+    function delegateBridge(
+        address to,
+        uint256 chainId,
+        address token,
+        uint256 amount,
+        SwapQuery memory destQuery
+    ) external payable {
```

Comment:
Any thoughts on removing `payable` given the delegate call? Unclear to me we need it for the bridge module and reminds me of this issue every time I see it https://github.com/Uniswap/v3-periphery/issues/52 (although not the same as multicall here)

Reply:
We need the `payable` here, as the RouterV2 main entry function is payable (user can start from native ETH). In the context of delegate call module and Router are sharing `msg.value`, so removing `payable` leads to failed tests:
```
[FAIL. Reason: Expected an emit, but the call reverted instead. Ensure you're testing the happy path when using the `expectEmit` cheatcode.] testDelegateBridgeMsgValueDeposit() (gas: 454483)
Traces:
  [454483] SynapseBridgeModuleTest::testDelegateBridgeMsgValueDeposit()
    ├─ ...
    ├─ [5090] DelegateCaller::performDelegateCall{value: 2000000000000000000}(SynapseBridgeModule: [0x5991A2dF15A8F6A256D3Ec51E99254Cd3fb576A9], 0x436f3aa5000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000014000000000000000000000000c7183455a4c133ae270771860664b6b7ec320bb10000000000000000000000000000000000000000000000000de0b6b3a764000000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000) 
    │   ├─ [45] SynapseBridgeModule::delegateBridge(0x000000000000000000000000000000000000000A, 20, DT: [0xc7183455a4C133Ae270771860664b6B7ec320bB1], 1000000000000000000 [1e18], (0x0000000000000000000000000000000000000000, 0x0000000000000000000000000000000000000000, 0, 0, 0x)) [delegatecall]
    │   │   └─ ← "EvmError: Revert"
    │   └─ ← "Address: low-level delegate call failed"
```

Reply:
Understood. But thought at the router level user is supposed to use the adapter to deposit native ETH to WETH prior to interacting with bridge?